package/MDAnalysis/analysis/dssp.py

```diff
@@ -0,0 +1,279 @@
+    vec_can = coord[:, 1:, 0] - coord[:, 1:, 1]
+    vec_can = vec_can / np.linalg.norm(vec_can, axis=-1, keepdims=True)
+    vec_nh = vec_cn + vec_can
+    vec_nh = vec_nh / np.linalg.norm(vec_nh, axis=-1, keepdims=True)
+    return coord[:, 1:, 0] + 1.01 * vec_nh
+
+
+def get_hbond_map(
+    coord: np.ndarray,
+    cutoff: float = DEFAULT_CUTOFF,
+    margin: float = DEFAULT_MARGIN,
+    return_e: bool = False,
+) -> np.ndarray:
+    """Returns hydrogen bond map
+
+    Args:
+        coord (np.ndarray): input coordinates in either (n, 4, 3) or (n, 5, 3) shape (without or with hydrogens)
+        cutoff (float, optional): Defaults to DEFAULT_CUTOFF.
+        margin (float, optional): margin. Defaults to DEFAULT_MARGIN.
+        return_e (bool, optional): if to return energy instead of hbond map. Defaults to False.
+
+    Returns:
+        np.ndarray: output hbond map or energy depending on return_e param
+    """
+    # check input
+    coord, org_shape = _check_input(coord)
+    b, l, a, _ = coord.shape
+    # add pseudo-H atom if not available
+    assert (a == 4) or (
+        a == 5
+    ), "Number of atoms should be 4 (N,CA,C,O) or 5 (N,CA,C,O,H)"
+    h = coord[:, 1:, 4] if a == 5 else _get_hydrogen_atom_position(coord)
+    # distance matrix
+    nmap = repeat(coord[:, 1:, 0], "... m c -> ... m n c", n=l - 1)
+    hmap = repeat(h, "... m c -> ... m n c", n=l - 1)
+    cmap = repeat(coord[:, 0:-1, 2], "... n c -> ... m n c", m=l - 1)
+    omap = repeat(coord[:, 0:-1, 3], "... n c -> ... m n c", m=l - 1)
+    d_on = np.linalg.norm(omap - nmap, axis=-1)
+    d_ch = np.linalg.norm(cmap - hmap, axis=-1)
+    d_oh = np.linalg.norm(omap - hmap, axis=-1)
+    d_cn = np.linalg.norm(cmap - nmap, axis=-1)
+    # electrostatic interaction energy
+    e = np.pad(
+        CONST_Q1Q2 * (1.0 / d_on + 1.0 / d_ch - 1.0 /
+                      d_oh - 1.0 / d_cn) * CONST_F,
+        [[0, 0], [1, 0], [0, 1]],
+    )
+    if return_e:
+        return e
+    # mask for local pairs (i,i), (i,i+1), (i,i+2)
+    local_mask = ~np.eye(l, dtype=bool)
+    local_mask *= ~np.diag(np.ones(l - 1, dtype=bool), k=-1)
+    local_mask *= ~np.diag(np.ones(l - 2, dtype=bool), k=-2)
+    # hydrogen bond map (continuous value extension of original definition)
+    hbond_map = np.clip(cutoff - margin - e, a_min=-margin, a_max=margin)
+    hbond_map = (np.sin(hbond_map / margin * np.pi / 2) + 1.0) / 2
+    hbond_map = hbond_map * repeat(local_mask, "l1 l2 -> b l1 l2", b=b)
+    # return h-bond map
+    hbond_map = np.squeeze(hbond_map, axis=0) if len(
+        org_shape) == 3 else hbond_map
+    return hbond_map
+
+
+def assign(coord: np.ndarray) -> np.ndarray:
+    """Assigns secondary structure for a given coordinate array
+
+    Args:
+        coord (np.ndarray): input coordinates in either (n, 4, 3) or (n, 5, 3) shape -- without or with hydrogens
+
+    Returns:
+        np.ndarray: output (n,) array with labels in C3 notation ('-', 'H', 'E')
```

Comment:
numpy doc

Reply:
Does this version of the algorithm only recognize helix and sheet?

Is "helix" only alpha-helix? Or does it also contain 3_10 and π helix?

Document the limitation.

Reply:
I added the documentation to the class itself -- seems that all helices go together, as well as all sheets.

Btw, it seems that the code already can detect pi- and 3-10 helices: https://github.com/ShintaroMinami/PyDSSP/blob/9bf4034c1584864eabd9aa93ed5227d791ff71d9/pydssp/pydssp_numpy.py#L100-L107

But I believe it'd require some benchmarking, which I'm not sure if I can do.